bgpd/bgp_io.h

```diff
@@ -22,8 +22,10 @@
 #ifndef _FRR_BGP_IO_H
 #define _FRR_BGP_IO_H
 
-#define BGP_WRITE_PACKET_MAX 64U
-#define BGP_READ_PACKET_MAX  10U
+#define BGP_WRITE_PACKET_DEFAULT 64U
+#define BGP_READ_PACKET_DEFAULT 10U
+#define BGP_WRITE_PACKET_MAX 255U
+#define BGP_READ_PACKET_MAX 255U
```

Comment:
This seems like a very risky change, in a couple of ways:

the allocation of the receive ringbuffer is current fixed - it's possible to compute exactly how much memory will be used because the packet size (64K) and number of packets (10) are fixed. changing the limit by a factor of 25 makes it possible to increase the memory footprint by a factor of 25 - right? and that might be disastrous - there's no check, for example, that the configuration change could actually be supported by available memory in the system. if the administrator changes the config, the box could a) begin crashing when malloc fails, or b) start paging, crushing performance and making any scalability issues even worse.

2. It already seems to me that there's a scaling problem with incoming reads: the buffer available for reads scales linearly with the number of connections - but the resources to process incoming packets, in the main pthread, is constant. adding more read-side buffering and making it even easier to overrun the main pthread seems hard to understand - I'd want to really see something meaningful about how that could possibly help.

Reply:
thanks Mark, give me some more time to elaborate an answer.